src/main/java/net/apnic/whowas/rdap/AutNum.java

```diff
@@ -1,46 +1,100 @@
 package net.apnic.whowas.rdap;
 
 import net.apnic.whowas.history.ObjectKey;
-import net.apnic.whowas.rpsl.RpslObject;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
 
 /**
- * Static method for constructing AutNum objects from WHOIS aut-num records
+ * AutNum RDAP object.
  */
-class AutNum extends AbstractRdapObject{
-    private static long parseAutNum(ObjectKey objectKey) {
-        int dotIndex = objectKey.getObjectName().indexOf(".");
-        if (dotIndex > 0) {
-            return (Long.parseLong(objectKey.getObjectName().substring(2, dotIndex)) << 16)
-                    + Long.parseLong(objectKey.getObjectName().substring(dotIndex + 1));
-        } else {
-            return Long.parseLong(objectKey.getObjectName().substring(2));
+public class AutNum
+    extends GenericObject
```

Comment:
{comment} toString methods for the generic objects (autnum, domain, entity, ip network) could be useful, if only for debugging. (comment not suggestion)

Reply:
Yes agreed. Will have a look into this

Reply:
Done & Done